pkg/auth/service.go

```diff
@@ -1040,19 +1011,39 @@ func (s *DBAuthService) ClaimTokenIDOnce(ctx context.Context, tokenID string, ex
 }
 
 // markTokenSingleUse returns true if token is valid for single use
-func (s *DBAuthService) markTokenSingleUse(ctx context.Context, tokenID string, tokenExpiresAt time.Time) (bool, error) {
-	res, err := s.db.Exec(ctx, `INSERT INTO auth_expired_tokens (token_id, token_expires_at) VALUES ($1,$2) ON CONFLICT DO NOTHING`,
-		tokenID, tokenExpiresAt)
+func (s *KVAuthService) markTokenSingleUse(ctx context.Context, tokenID string, tokenExpiresAt time.Time) (bool, error) {
+	tokenPath := model.KvTokenPath(tokenID)
+	m := model.TokenData{TokenId: tokenID, ExpiredAt: timestamppb.New(tokenExpiresAt)}
+	err := s.store.SetMsg(ctx, model.PartitionKey, tokenPath, &m)
 	if err != nil {
 		return false, err
 	}
-	canUseToken := res.RowsAffected() == 1
+
 	// cleanup old tokens
-	_, err = s.db.Exec(ctx, `DELETE FROM auth_expired_tokens WHERE token_expires_at < $1`, time.Now())
+	t := model.TokenData{}
```

Comment:
This is where we differ from DB. I don't think this is the appropriate place for O(n) cleanups. DB cleanups are O(1) and therefore can be done here.

Reply:
opened issue #3500